src/betterproto/__init__.py

```diff
@@ -1178,61 +1217,40 @@ def to_dict(
                     output[cased_name] = value
         return output
 
-    def from_dict(self: T, value: Mapping[str, Any]) -> T:
-        """
-        Parse the key/value pairs into the current message instance. This returns the
-        instance itself and is therefore assignable and chainable.
-
-        Parameters
-        -----------
-        value: Dict[:class:`str`, Any]
-            The dictionary to parse from.
-
-        Returns
-        --------
-        :class:`Message`
-            The initialized message.
-        """
-        self._serialized_on_wire = True
+    @classmethod
+    def _from_dict_init(cls: Type[T], value: Mapping[str, Any]) -> Mapping[str, Any]:
+        metadata = ProtoClassMetadata(cls)
```

Comment:
This is not the correct way to work around this problem. ProtoClassMetadata should be made a cached class property.